python_modules/libraries/dagster-dbt/dagster_dbt/dbt_artifacts.py

```diff
@@ -0,0 +1,162 @@
+        * During development, reload the manifest at run time to pick up any changes.
+        * When deploying, expect a manifest that was prepared at build time to reduce start-up time.
+        * Handle the scenario when the dbt project is copied in to a directory for packaging.
+
+        Args:
+            project_dir (Union[str, Path]):
+                The directory of the dbt project.
+            target_folder (Union[str, Path]):
+                The folder in the project project directory to output artifacts.
+                Default: "target"
+            prepare_command: The dbt cli command to run to prepare the manifest.json
+                Default: ["parse", "--quiet"]
+            package_data_dir (Union[str, Path]):
+                A directory that will contain a copy of the dbt project, synced at build time.
+        """
+        self.project_dir = Path(project_dir)
+        if not self.project_dir.exists():
+            raise DagsterDbtProjectNotFoundError(f"project_dir {project_dir} does not exist.")
+        self._target_folder = Path(target_folder)
+        self._prepare_command = prepare_command
+        # this is ok if it doesn't exist, will get created by `prepare`
+        self._package_data_dir = Path(package_data_dir) if package_data_dir else None
+
+        if self._should_prepare_at_runtime():
+            self._manifest_path = self._prepare_manifest()
+        else:
+            self._manifest_path = self._base_dir.joinpath(self._target_folder, "manifest.json")
+
+    @public
+    @property
+    def manifest_path(self) -> Path:
+        """The path to the manifest.json, compiling the manifest first if in development or
+        ensuring it already exists if not.
+        """
+        if not self._manifest_path.exists():
+            if self._should_prepare_at_runtime():
+                raise DagsterDbtManifestNotPreparedError(
+                    f"Unexpected state, {self._manifest_path} should have been created when this "
+                    "object was initialized."
+                )
+            else:
+                raise DagsterDbtManifestNotPreparedError(
+                    f"Did not find prepared manifest.json at expected path {self._manifest_path}.\n"
+                    "If this is an environment that is expected to prepare the manifest at run time, "
+                    "set the environment variable DAGSTER_DBT_PARSE_PROJECT_ON_LOAD."
+                )
+
+        return self._manifest_path
+
+    @public
+    def prepare(self, *, quiet=False) -> None:
+        """A method that can be called as part of the deployment process to handle
+        preparing the manifest and if package_data_dir is set, handle copying
+        the dbt project to that directory.
+
+        Args:
+            quiet (bool):
+                Disable logging
+                Default: False
+        """
+        logger = _NULL_LOGGER if quiet else logging.getLogger("dagster-dbt-artifacts")
```

Comment:
I think there was more discussion elsewhere, but any particular reason for not logging at all instead of just logging at DEBUG level vs. INFO level? I feel like that's a more common pattern in our codebase.

No strong feelings there just trying to understand.